### PR TITLE
MON-2642: Improve e2e tests for alertrelabelconfigs CRD

### PR DESCRIPTION
**Pull request by machine424**
Check on Prometheus config secret instead of relabelConfig secret to test more things/parts.

* [ ] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

**Comment by danielmellado** [comment]
Nice addition of testify over there, my only concern would be that the usage of this is homogeneous all along the tests. I'd say yes to this and consider this a start PR ;)

File: test/e2e/framework/prometheus.go [hunk 1]
```diff
@@ -133,3 +140,20 @@ func (f Framework) MakePrometheusServiceRoute(svc *v1.Service) *routev1.Route {
 		},
 	}
 }
+
+func (f Framework) PrometheusConfigFromSecret(t *testing.T, namespace, secretName string) (*promConfig.Config, error) {
+	rawConfig := f.MustGetSecret(t, secretName, namespace).Data["prometheus.yaml.gz"]
```
Comment (simonpasquier) [inline test/e2e/framework/prometheus.go:145]:
```suggestion
	t.Helper()

	rawConfig := f.MustGetSecret(t, secretName, namespace).Data["prometheus.yaml.gz"]
```

File: test/e2e/alert_relabel_config_test.go [hunk 2]
```diff
@@ -56,124 +61,70 @@ func TestAlertRelabelConfigTechPreview(t *testing.T) {
...
-
-		return nil
-	})
+	// Check Prometheus config forgot about the deleted AlertRelabelConfig.
+	validateCurrentRelabelConfig(t, initialRelabelConfig)
+}
 
+// prometheusRelabelConfig returns the alert relabel configuration part used by Prometheus config
+func prometheusRelabelConfig(t *testing.T) ([]*relabel.Config, error) {
+	prometheusConfig, err := f.PrometheusConfigFromSecret(t, f.Ns, prometheusConfigSecretName)
```
Comment (simonpasquier) [inline test/e2e/alert_relabel_config_test.go:107]: If the function gets a t parameter, we can check the error directly?
```suggestion
	t.Helper()

	prometheusConfig, err := f.PrometheusConfigFromSecret(t, f.Ns, prometheusConfigSecretName)
	require.NoError(t, err)
```

File: test/e2e/alert_relabel_config_test.go [hunk 3]
```diff
@@ -56,124 +61,70 @@ func TestAlertRelabelConfigTechPreview(t *testing.T) {
...
-		return fmt.Errorf("TargetLabel does not match (%q != %q)", arc.TargetLabel, fromSecret.TargetLabel)
-	}
+		if !assert.ElementsMatch(t, expectedRelabelConfig, currentRelabelConfig) {
+			return errors.New("the expected relabel config is not applied yet.")
+		}
+		return nil
+	})
 
-	arcRegex, err := relabel.NewRegexp(arc.Regex)
 	if err != nil {
```
Comment (simonpasquier) [inline test/e2e/alert_relabel_config_test.go:127]: (nit) could use `require.NoError(t, err)`?

**Review by simonpasquier — COMMENTED** [review]
Will the test also check that the resulting alert in Alertmanager is modified as expected?

File: test/e2e/framework/prometheus.go (hunk 1 above)
Comment (machine424) [reply test/e2e/framework/prometheus.go:146]: Cool for the `t.Helper()`

File: test/e2e/alert_relabel_config_test.go (hunk 3 above)
Comment (machine424) [reply test/e2e/alert_relabel_config_test.go:123]: Now that there is no objection to testify, I migrated this and some others.

File: test/e2e/alert_relabel_config_test.go (hunk 2 above)
Comment (machine424) [reply test/e2e/alert_relabel_config_test.go:103]: I thought about returning an error to be able to retry this as well if there are any intermittent problems, do you think we should be aware of those and fail from the first time?

Comment (simonpasquier) [reply test/e2e/alert_relabel_config_test.go:103]: hmm right after reading again `PrometheusConfigFromSecret()` it doesn't seem to use to the `t` parameter so probably the best option is to remove it from the signature?

Comment (machine424) [reply test/e2e/alert_relabel_config_test.go:103]: Actually we pass it to `MustGetSecret` and we use it for `t.Helper()` now.

Comment (simonpasquier) [reply test/e2e/alert_relabel_config_test.go:103]: Ah yes. Then maybe `PrometheusConfigFromSecret()` shouldn't return an error? If the secret data is invalid (e.g. can't be unzipped/deserialized), we should fail hard with `t.Fail`?

Comment (machine424) [reply test/e2e/alert_relabel_config_test.go:103]: That was my question :) I see that `MustGetSecret` already has retries that should deal with network/apiserver `intermittent problems`, we're left with unzipping/deserialization problems, which for sure we don't want to hide/we can never hide. Done.

Comment (simonpasquier) [reply test/e2e/alert_relabel_config_test.go]: :+1:

**Review by simonpasquier — COMMENTED** [review]
/lgtm

**Comment by simonpasquier** [comment]
/label acknowledge-critical-fixes-only

no impact on the release payload.

**Pull request merged**